flume-ng-sources/flume-kafka-source/src/main/java/org/apache/flume/source/kafka/KafkaSource.java

```diff
@@ -515,6 +530,14 @@ Properties getConsumerProps() {
     return stringMap;
   }
 
+  private static Map<String, String> toStringMap(Headers kafkaHeaders) {
+    Map<String, String> stringMap = new HashMap<String, String>();
+    for (Header kafkaHeader : kafkaHeaders) {
+      stringMap.put(kafkaHeader.key(), new String(kafkaHeader.value()));
```

Comment:
Should be construct through `public String(byte bytes[], String charsetName)` with a configurable Charset value